components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/impl/ActionExecutorServiceImpl.java

```diff
@@ -265,6 +277,30 @@ private ActionExecutionStatus<?> executeAction(Action action,
         }
     }
 
+    private boolean evaluateActionRule(Action action, Map<String, Object> eventContext, String tenantDomain)
+            throws ActionExecutionException {
+
+        if (action.getActionRule() != null && action.getActionRule().getId() != null) {
```

Comment:
switching the if condition as below will reduce the lines of code that is nested.

```
   if (action.getActionRule() == null && action.getActionRule().getId() == null) {
        logNoRuleConfiguredForAction(action);
        return true;
   }
```

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6324/commits/83c62ef36d70cefe80dd965a0fbcd3f4b6c67ed9